source/engine/graphics/system/EnvironmentSystem.h

```diff
@@ -1,12 +1,28 @@
 #pragma once
 
-#include "graphics/shader_resource/EnvironmentData.h"
+#include "graphics/shader_resource/ShaderResourceBus.h"
+#include "graphics/shader_resource/UniformBufferHandle.h"
+#include "graphics/shader_resource/TextureArrayBufferHandle.h"
 #include "utility/Signal.h"
 
+#include "ncmath/Vector.h"
+
 namespace nc::graphics
 {
 struct CameraState;
 
+#ifdef _MSC_VER
+#pragma warning( disable : 4324 ) // Justification: We want to be explicit about alignment for the GPU buffer structs.
+#endif
+struct EnvironmentData
+{
+    alignas(16)Vector4 cameraWorldPosition = Vector4(0.0f, 4.0f, -6.4f, 0.0f);
+    alignas(16)uint32_t skyboxTextureIndex = std::numeric_limits<uint32_t>::max();
```

Comment:
Holy cow, do all scalar types need 16 byte alignment?

Reply:
No - https://registry.khronos.org/vulkan/specs/1.3-extensions/html/chap15.html#interfaces-resources-layout
I'm still shaky on alignment, but I do know that we should avoid Vector3 and always use Vector4. Since Vector4 has an alignment of 16 - I thought that all members should have an alignment of 16.

Reply:
Ok, should be fine to give it extra alignment here, but FYI, both `nc::Vector3` and `nc::Vector4` have 4 byte alignment, not 16. `DirectX::XMVECTOR` is 16, though. Unresolving just so you see this, in case that matters anywhere.